### PR TITLE
[KBFS-3230] widget chat container

### PR DESCRIPTION
**Pull request by jzila**
I don't want this merged until we have some chat content for the box, which will come in a separate PR.

Ready for feedback though.

### CODE REVIEW

File: shared/menubar/chat.desktop.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      marginTop: Styles.globalMargins.xtiny,
+      paddingBottom: Styles.globalMargins.xtiny,
+      paddingTop: Styles.globalMargins.xtiny,
+    },
+    isElectron: {
+      marginLeft: Styles.globalMargins.tiny,
+      marginRight: Styles.globalMargins.tiny,
+      paddingLeft: Styles.globalMargins.tiny,
+      paddingRight: Styles.globalMargins.tiny,
+    },
```
Comment (songgao) [inline shared/menubar/chat.desktop.js]: Is there a reason why you split them into common and isElectron? It only runs on desktop.

File: shared/menubar/index.desktop.js [hunk 2]
```diff
@@ -277,133 +203,132 @@ class MenubarRender extends Component<Props, State> {
...
+      right: 0,
+      marginLeft: 'auto',
+      marginRight: 'auto',
+      top: -6,
+    },
+    isElectron: {
+      borderLeft: '6px solid transparent',
+      borderRight: '6px solid transparent',
+      borderBottom: `6px solid ${Styles.globalColors.darkBlue2}`,
+    },
```
Comment (songgao) [inline shared/menubar/index.desktop.js]: Hmm does flow complain if these weren't in isElectron?

File: shared/menubar/index.desktop.js [hunk 3]
```diff
@@ -277,133 +203,132 @@ class MenubarRender extends Component<Props, State> {
...
+    common: {
+      ...Styles.globalStyles.flexBoxColumn,
+      flex: 1,
+      position: 'relative',
+      marginTop: isDarwin ? 13 : 0,
+      borderTopLeftRadius: Styles.globalMargins.xtiny,
+      borderTopRightRadius: Styles.globalMargins.xtiny,
+      backgroundColor: Styles.globalColors.darkBlue,
+    },
+  }),
```
Comment (songgao) [inline shared/menubar/index.desktop.js]: These and some below only have `common`, so why not just do this instead?

```
widgetContainer: {
    ...Styles.globalStyles.flexBoxColumn,
    ... ...
},
```

File: shared/menubar/index.desktop.js [hunk 4]
```diff
@@ -80,95 +54,93 @@ class MenubarRender extends Component<Props, State> {
...
         title: 'Settings',
-        view: this._menuView('Settings', 'iconfont-nav-settings', countMap[settingsTab] || 0),
-        onClick: () => this.props.openApp(settingsTab),
+        view: this._menuView('Settings', 'iconfont-nav-settings', countMap[Tabs.settingsTab] || 0),
+        onClick: () => this.props.openApp(Tabs.settingsTab),
       },
       'Divider',
       ...(this.props.loggedIn ? [{title: 'Open main app', onClick: () => this.props.openApp()}] : []),
-      {title: 'Open folders', onClick: () => this.props.openApp(fsTab)},
+      {title: 'Open folders', onClick: () => this.props.openApp(Tabs.fsTab)},
```
Comment (songgao) [inline shared/menubar/index.desktop.js]: Should this be "Open Files"?

Comment (jzila) [reply shared/menubar/index.desktop.js]: The design says "Open folders", but I think you're right, it should be "files".

File: shared/menubar/index.desktop.js (hunk 2 above)
Comment (jzila) [reply shared/menubar/index.desktop.js]: Yes

File: shared/menubar/index.desktop.js (hunk 3 above)
Comment (jzila) [reply shared/menubar/index.desktop.js]: Mainly because of Flow.

File: shared/menubar/chat.desktop.js (hunk 1 above)
Comment (jzila) [reply shared/menubar/chat.desktop.js]: Yes, Flow.

File: shared/menubar/index.desktop.js (hunk 2 above)
Comment (chrisnojima) [reply shared/menubar/index.desktop.js]: If you use the long form it's compatible

Comment (jzila) [reply shared/menubar/index.desktop.js]: Long form?

Comment (chrisnojima) [reply shared/menubar/index.desktop.js]: This is the short hand form. If you do borderLeftWidth: 6   BorderStyle: 'solid' borderLeftColor etc they're the same. Rn doesn't support the short hand where you can define multiple props at the same time

Comment (chrisnojima) [reply shared/menubar/index.desktop.js]: https://developer.mozilla.org/en-US/docs/Web/CSS/Shorthand_properties

**Comment by jzila** [comment]
PR feedback implemented, ready for another look.

File: shared/menubar/index.desktop.js [hunk 5]
```diff
@@ -140,7 +140,7 @@ class MenubarRender extends React.Component<Props, State> {
       },
       'Divider',
       ...(this.props.loggedIn ? [{title: 'Open main app', onClick: () => this.props.openApp()}] : []),
-      {title: 'Open folders', onClick: () => this.props.openApp(Tabs.fsTab)},
+      {title: 'Open files', onClick: () => this.props.openApp(Tabs.fsTab)},
```
Comment (songgao) [inline shared/menubar/index.desktop.js]: I wonder if this should be `Open Files`. `Files` isn't a word but the name of a tab here, so I don't think it'd contradict to Cécile's preference of not capitalize every word. No strong opinion though, so feel free to leave it!

Comment (jzila) [reply shared/menubar/index.desktop.js]: I'm going to leave it for now. We can always change it later.

**Pull request merged**